src/stores/appStore.ts

```diff
@@ -9,101 +8,63 @@ import {
-                .map(([, indicators]) => indicators);
-
-            return Object.assign({}, admin2Indicators[selectedCountryId], ...filteredAdmin1);
-        },
-        selectedFeatures: (state): Feature[] => {
-            const { selectedCountryId, admin1Geojson, admin2Geojson } = state;
-            // get single array of all selected features
-            if (!selectedCountryId) {
-                return Object.values(admin1Geojson).flatMap((geojson) => geojson);
-            }
-
-            const filteredAdmin1 = Object.entries(admin1Geojson)
-                .filter(([countryId]) => countryId !== selectedCountryId)
-                .flatMap(([, geojson]) => geojson);
-
-            return [...admin2Geojson[selectedCountryId], ...filteredAdmin1];
-        }
-    },
     actions: {
         async initialiseData() {
-            this.appConfig = await getAppConfig();
-
-            // Read all adm1 indicators from a single file
-            const allIndicators = await getGlobalIndicators(1);
-            // Load all admin1 geojson - load simplified boundaries only, from a
-            // single file,
-            const allGeojson = await getGlobalGeojsonFeatures(1);
+            const state = this as AppState;
+            state.appConfig = await getAppConfig();
+            state.countryBoundingBoxes = await getCountryBoundingBoxes();
+            state.admin1Geojson = await getGlobalGeojsonFeatures(1);
+            state.admin1Indicators = await getGlobalIndicators(1);
 
-            this.countryBoundingBoxes = await getCountryBoundingBoxes();
-            this.admin1Geojson = allGeojson;
-            this.admin1Indicators = allIndicators;
+            state.initialisationComplete = true;
         },
-        async selectCountry(countryId: string) {
-            if (countryId === this.selectedCountryId) {
-                return;
-            }
+        async updateMapSettings(newMapSettings: MapSettings) {
+            const state = this as AppState;
+            const { country, adminLevel } = newMapSettings;
 
-            if (!countryId) {
-                this.selectedCountryId = "";
-                this.admin0GeojsonFeature = null;
+            if (!country) {
+                state.mapSettings = newMapSettings;
+                state.admin0GeojsonFeature = null;
                 return;
             }
 
+            let level = adminLevel;
             // Some countries do not have admin2 regions or data - if one of these is selected, we load
             // a more detailed geojson, and re-use its admin1 indicators as "admin2"
-            const level = this.appConfig.countriesWithoutAdmin2.includes(countryId) ? 1 : 2;
+            const admin2DataMissing = state.appConfig.countriesWithoutAdmin2.includes(country);
+            if (level === 2 && admin2DataMissing) {
+                level = 1;
+                state.admin2Indicators[country] = state.admin1Indicators[country];
```

Comment:
I see that you didn't introduce this in this PR, but it makes me uneasy to store admin1 indicators under the name 'admin2', for reasons of making how things work more confusing. Maybe I'd prefer that we don't offer the 'admin 2' option to users if they have selected a country without admin 2 regions. If people agree with that, I'll make a ticket for it.

Reply:
The selected admin level is `1` if a country has missing admin level 2 data, that is what the `finalMapSettings` line does in this function so i think in the future when the admin level toggle is added it will show admin level 1 features only, but i agree that there could be a potentially better method to doing this than putting the admin 1 indicators into admin 2 indicators